Source/WebKit/NetworkProcess/ServiceWorker/WebSWServerConnection.cpp

```diff
@@ -386,17 +387,8 @@ void WebSWServerConnection::scheduleUnregisterJobInServer(ServiceWorkerJobIdenti
 
 void WebSWServerConnection::postMessageToServiceWorkerClient(ScriptExecutionContextIdentifier destinationContextIdentifier, const MessageWithMessagePorts& message, ServiceWorkerIdentifier sourceIdentifier, const String& sourceOrigin)
 {
-    auto* sourceServiceWorker = server().workerByID(sourceIdentifier);
-    if (!sourceServiceWorker)
-        return;
-
-    auto sourceServiceWorkerData = sourceServiceWorker->data();
-
-    sendWithAsyncReply(Messages::WebSWClientConnection::PostMessageToServiceWorkerClient { destinationContextIdentifier, message, sourceServiceWorkerData, sourceOrigin }, [this, weakThis = WeakPtr { *this }, destinationContextIdentifier, message, sourceServiceWorkerData, sourceOrigin] (bool result) {
-        if (result)
-            return;
-
-        server().addServiceWorkerClientPendingMessage(destinationContextIdentifier, { message, sourceServiceWorkerData, sourceOrigin });
+    server().postMessageToServiceWorkerClient(destinationContextIdentifier, message, sourceIdentifier, sourceOrigin, [this] (auto destinationContextIdentifier, auto& message, auto sourceServiceWorkerData, auto& sourceOrigin) {
+        send(Messages::WebSWClientConnection::PostMessageToServiceWorkerClient { destinationContextIdentifier, message, sourceServiceWorkerData, sourceOrigin }, 0);
```

Comment:
What guarantees that `this` is still alive here?

Reply:
We pass a const Function& so that can only be called synchronously.
I probably need to refactor a bit this code anyway to handle the case of postMessaging to a PSON service worker client. This will probably remove this lambda.